DocumentsFromSnapshotMigration/src/test/java/com/rfs/FullTest.java

```diff
@@ -119,13 +120,15 @@ public void testDocumentMigration(SearchClusterContainer.Version baseSourceImage
                      String targetImageName, int numWorkers)
             throws Exception
     {
-
+        var executorService = Executors.newFixedThreadPool(numWorkers);
         try (var esSourceContainer = new PreloadedSearchClusterContainer(baseSourceImageVersion,
                 SOURCE_SERVER_ALIAS, generatorImage, generatorArgs);
              OpensearchContainer<?> osTargetContainer =
                      new OpensearchContainer<>(targetImageName)) {
-            esSourceContainer.start();
-            osTargetContainer.start();
+            CompletableFuture.allOf(
+                CompletableFuture.supplyAsync(()->{ esSourceContainer.start(); return null; }, executorService),
```

Comment:
Nit: This compiles in my IDE if you can make it happen here and the following line
```suggestion
                CompletableFuture.supplyAsync(()->esSourceContainer.start(), executorService),
```

Reply:
Alternatively proposal, refactor the container starts into a reused function between test cases